libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -203,7 +211,8 @@ void WasmDialect::addEthereumExternals()
 		f.sideEffects = SideEffects::worst();
 		f.controlFlowSideEffects = ext.controlFlowSideEffects;
 		f.isMSize = false;
-		f.sideEffects.invalidatesStorage = (ext.name == "storageStore");
+		f.sideEffects.storage = (ext.name == "storageStore") ? SideEffects::Write : SideEffects::Pure;
```

Comment:
Should have `reod` for `storageLoad` (please look up the correct name, though)

Reply:
There is a TODO on L210, about adding side effects properly. Do you think I should do it here?

Reply:
just add the side effects for the builtins we already have here